### PR TITLE
ci-e2e-upgrade: Make it stable

### PR DESCRIPTION
**Pull request by brb**
See commit msgs.

Successful run - https://github.com/cilium/cilium/actions/runs/8647377764.

### CODE REVIEW

**Review by brlbil — APPROVED** [review]

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 1]
```diff
@@ -158,6 +160,8 @@ jobs:
             lb-mode: 'snat'
             egress-gateway: 'true'
             lb-acceleration: 'testing-only'
+            # Disable until https://github.com/cilium/cilium/issues/30717
+            # has been resolved.
             ingress-controller: 'true'
```
Comment (joestringer) [inline .github/workflows/tests-e2e-upgrade.yaml:165]: This doesn't disable ingress-controller. :thinking:

Comment (joestringer) [reply .github/workflows/tests-e2e-upgrade.yaml:165]: There's several more cases like this. cc @brlbil between author and reviewer, we should be catching this stuff. At least for me, this becomes immediately apparent when reviewing commit-by-commit.

Comment (brlbil) [reply .github/workflows/tests-e2e-upgrade.yaml:165]: @joestringer you are right. It seems that the midnight review is not a good idea after all.

Comment (joestringer) [reply .github/workflows/tests-e2e-upgrade.yaml:165]: :sweat_smile: Live and learn. Looks like @brb is already on top of it: https://github.com/cilium/cilium/pull/31917

**Pull request merged**